doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,250 @@
+7. Wait for at least 65 seconds. Verify target speed of all FANs are set to 60% according to valid_policy.json.
+8. Make fake data: first FAN speed exceed threshold(speed < target speed), second FAN speed exceed threshold(speed > target speed).
+9. Wait for at least 65 seconds. Verify led turns to red for first and second FAN.
+10. Make fake data: first and second FAN speed recover to normal.
+11. Wait for at least 65 seconds. Verify led turns to green for first and second FAN.
+12. Link FAN related sysfs. Restore the original policy file.
+
+### PSU Absence Test
+
+PSU absence test verifies that once any PSU absence, all FAN speed will be set to proper value according to policy file.
+
+#### Procedure
+
+1. Testbed setup.
+2. Copy valid_policy.json to pmon docker and backup the original one.
+3. Restart pmon service to trigger thermal control daemon reload policy configuration file.
+4. Turn off two PSUs.
+5. Wait for at least 65 seconds. Verify target speed of all FANs are set to 100% according to valid_policy.json.
+6. Turn on one PSU.
+7. Wait for at least 65 seconds. Verify target speed of all FANs are still 100% because there is still one PSU absence.
+8. Turn on all PSU.
+9. Verify target speed of all Fans are set to 60% according to valid_policy.json.
+10. Restore the original policy file.
+
+> Note: The reason that we wait at least 65 seconds is that thermal policy run every 60 seconds according to design.
+> For switch who has only one PSU, step 6 and step 7 will be ignored.
+
+### Invalid Policy Format Load Test
+
+Invalid policy format test verifies that thermal control daemon does not exit when loading a invalid_format_policy.json file. The thermal control daemon cannot perform any thermal policy in this case, but FAN monitor and thermal monitor should still work.
+
+#### Procedure
+
+1. Testbed setup.
+2. Copy invalid_format_policy.json to pmon docker and backup the original one.
+3. Restart pmon service to trigger thermal control daemon reload policy configuration file.
+4. Verify FAN status and thermal status still display normally. (By check command output of `show platform fanstatus` and `show platform temperature`)
+5. Recover the original policy configuration file and restart pmon service
+
+### Invalid Policy Value Load Test
+
+Invalid policy value test verifies that thermal control daemon does not exit when loading a invalid_value_policy.json file. The thermal control daemon cannot perform any thermal policy in this case, but FAN monitor and thermal monitor should still work.
+
+#### Procedure
+
+1. Testbed setup.
+2. Copy invalid_value_policy.json to pmon docker and backup the original one.
+3. Restart pmon service to trigger thermal control daemon reload policy configuration file.
+4. Verify FAN status and thermal status still display normally. (By check command output of `show platform fanstatus` and `show platform temperature`)
+5. Recover the original policy configuration file and restart pmon service
```

Comment:
one thought about the restore,  maybe can combine the restore of each test case at the end of the test by a config reload, but this depends on how you implement the restore and whether the previous mock symbol link will impact the followed test.

Reply:
To unlink: just remove the symbol link and create a file with the same name.
To restore: remove the created file and add a soft link to hardware management file.
No need config reload.

Reply:
if you are going to restore by re linking, it requires to restore the link ahead of removing it.

Reply:
Yes. I will save the original link in the test case.